### PR TITLE
Initial PR

### PR DESCRIPTION
**Pull request by prerana1821**

### CODE REVIEW

File: components/ShowPassword/ShowPassword.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  showPassword,
+  showConfirmPassword,
+}) => {
+  return (
+    <button
+      type='button'
+      className='btnIcon btnIconShowPassword'
+      onClick={showPasswordHandler}
+    >
+      {showPassword ? (
```
Comment (uditkumar01) [inline components/ShowPassword/ShowPassword.js:14]: No need to use two `Image` components. Outer ternary and the ternary operator inside the `src` prop of the `Image` component are doing the same thing. Please check it once.

File: components/UserInterviewSlot/UserInterviewSlot.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import cn from "classnames";
```
Comment (uditkumar01) [inline components/UserInterviewSlot/UserInterviewSlot.js:1]: Please name it properly.

File: components/UserInterviewSlot/UserInterviewSlot.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  };
+
+  return (
+    <>
+      <h1 className='textCenter'>Your Interview Slots</h1>
+      <div className={userInterviewSlot.interviewSlots}>
+        {interviewSlotState?.userInterViewSlots?.slots?.map((item) => {
+          return (
+            <div
```
Comment (uditkumar01) [inline components/UserInterviewSlot/UserInterviewSlot.js:56]: You can make this a separate component as the current component already contains a lot of content.

File: context/AuthProvider.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  const router = useRouter();
+
+  const [authState, authDispatch] = useReducer(authReducer, {
+    token: "",
+    user: null,
+    status: null,
+  });
+
+  useEffect(() => {
+    (async () => {
```
Comment (uditkumar01) [inline context/AuthProvider.js:19]: extract this function in a separate file if possible.

File: context/AuthProvider.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+      const response = await fetch("/api/auth/signin", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ username, password, email }),
+      });
+
+      const data = await response.json();
+      console.log({ data });
```
Comment (uditkumar01) [inline context/AuthProvider.js:69]: remove console logs

File: pages/interviews.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        payload: {
+          status: { error: interviewSlots.message },
+        },
+      });
+    } else {
+      if (
+        Object.entries(interviewSlots).length === 0 ||
+        interviewSlots.length === 0
+      ) {
+        interviewSlotDispatch({
```
Comment (uditkumar01) [inline pages/interviews.js:27]: By default, you can keep the loading status. And then when you receive a response then make a dispatch call.

**Review by uditkumar01 — COMMENTED** [review]
Your code's readability is really good. I have mentioned few changes please check these.

**Comment by prerana1821** [comment]
> Your code's readability is really good. I have mentioned few changes please check these.

Hey udit, Thanks a lot for reviewing my code.
I have refactored the code as per the suggested changes!

**Pull request merged**